src/client/datascience/notebook/notebookControllerManager.ts

```diff
@@ -468,90 +406,4 @@ export class NotebookControllerManager implements INotebookControllerManager, IE
             controller.dispose();
         });
     }
-
-    private async notebookKernelChanged(document: NotebookDocument, controller: VSCodeNotebookController) {
```

Comment:
Moved all of this code into the controller, the controller has an event for whether its seleced or not, hence moved it closer to where the evnt is fired, instead of adding our own layer (our own event)..